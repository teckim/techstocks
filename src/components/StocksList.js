import React from 'react';
import PropTypes from 'prop-types';
import StockCard from './StockCard';

const StocksList = ({ stocks }) => (
  <div className="grid grid-cols-2">
    {
      stocks
        .filter((stock) => stock.country === 'US')
        .map((stock) => <StockCard key={stock.symbol} stock={stock} />)
    }
  </div>
);

StocksList.propTypes = {
  stocks: PropTypes.instanceOf(Array).isRequired,
};

export default StocksList;
