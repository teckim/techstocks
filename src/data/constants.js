export const FMG_API_KEY = process.env.REACT_APP_FMG_API_KEY;
export const API_ENDPOINT = 'https://financialmodelingprep.com/api/v3/';
