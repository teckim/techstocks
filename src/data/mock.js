export const company = {
  symbol: 'AAPL',
  price: 151.6,
  beta: 1.19455,
  volAvg: 91750000,
  mktCap: 2453676490752,
  lastDiv: 0.89,
  range: '129.04-182.94',
  changes: -1.3499908,
  companyName: 'Apple Inc.',
  currency: 'USD',
  cik: '0000320193',
  isin: 'US0378331005',
  cusip: '037833100',
  exchange: 'NASDAQ Global Select',
  exchangeShortName: 'NASDAQ',
  industry: 'Consumer Electronics',
  website: 'https://www.apple.com',
  description: 'Apple Inc. designs, manufactures, and markets smartphones, personal computers, tablets, wearables, and accessories worldwide. It also sells various related services. In addition, the company offers iPhone, a line of smartphones; Mac, a line of personal computers; iPad, a line of multi-purpose tablets; AirPods Max, an over-ear wireless headphone; and wearables, home, and accessories comprising AirPods, Apple TV, Apple Watch, Beats products, HomePod, and iPod touch. Further, it provides AppleCare support services; cloud services store services; and operates various platforms, including the App Store that allow customers to discover and download applications and digital content, such as books, music, video, games, and podcasts. Additionally, the company offers various services, such as Apple Arcade, a game subscription service; Apple Music, which offers users a curated listening experience with on-demand radio stations; Apple News+, a subscription news and magazine service; Apple TV+, which offers exclusive original content; Apple Card, a co-branded credit card; and Apple Pay, a cashless payment service, as well as licenses its intellectual property. The company serves consumers, and small and mid-sized businesses; and the education, enterprise, and government markets. It distributes third-party applications for its products through the App Store. The company also sells its products through its retail and online stores, and direct sales force; and third-party cellular network carriers, wholesalers, retailers, and resellers. Apple Inc. was incorporated in 1977 and is headquartered in Cupertino, California.',
  ceo: 'Mr. Timothy Cook',
  sector: 'Technology',
  country: 'US',
  fullTimeEmployees: '154000',
  phone: '14089961010',
  address: '1 Apple Park Way',
  city: 'Cupertino',
  state: 'CALIFORNIA',
  zip: '95014',
  dcfDiff: 2.07175,
  dcf: 153.672,
  image: 'https://financialmodelingprep.com/image-stock/AAPL.png',
  ipoDate: '1980-12-12',
  defaultImage: false,
  isEtf: false,
  isActivelyTrading: true,
  isAdr: false,
  isFund: false,
};

export const stocks = [
  {
    symbol: 'AAPL',
    companyName: 'Apple Inc.',
    marketCap: 2475526193152,
    sector: 'Technology',
    industry: 'Consumer Electronics',
    beta: 1.1946,
    price: 152.9500,
    lastAnnualDividend: 0.8900,
    volume: 92293137,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'MSFT',
    companyName: 'Microsoft Corporation',
    marketCap: 1935797125120,
    sector: 'Technology',
    industry: 'Software—Infrastructure',
    beta: 0.9323,
    price: 258.8300,
    lastAnnualDividend: 2.4200,
    volume: 30854719,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'NVDA',
    companyName: 'NVIDIA Corporation',
    marketCap: 424238088192,
    sector: 'Technology',
    industry: 'Semiconductors',
    beta: 1.6369,
    price: 170.2400,
    lastAnnualDividend: 0.1600,
    volume: 56788501,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'ASML',
    companyName: 'ASML Holding N.V.',
    marketCap: 214078865408,
    sector: 'Technology',
    industry: 'Semiconductor Equipment & Materials',
    beta: 1.1030,
    price: 538.9700,
    lastAnnualDividend: 5.0850,
    volume: 1102834,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'NL',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'AVGO',
    companyName: 'Broadcom Inc.',
    marketCap: 207798255616,
    sector: 'Technology',
    industry: 'Semiconductors',
    beta: 1.1047,
    price: 513.1100,
    lastAnnualDividend: 15.9000,
    volume: 2398488,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'CSCO',
    companyName: 'Cisco Systems, Inc.',
    marketCap: 183610163200,
    sector: 'Technology',
    industry: 'Communication Equipment',
    beta: 0.9621,
    price: 44.3400,
    lastAnnualDividend: 1.5000,
    volume: 24515332,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'ADBE',
    companyName: 'Adobe Inc.',
    marketCap: 183437279232,
    sector: 'Technology',
    industry: 'Software—Infrastructure',
    beta: 1.1081,
    price: 391.9600,
    lastAnnualDividend: 0.0000,
    volume: 3102967,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'QCOM',
    companyName: 'QUALCOMM Incorporated',
    marketCap: 171639996416,
    sector: 'Technology',
    industry: 'Semiconductors',
    beta: 1.2231,
    price: 153.2500,
    lastAnnualDividend: 2.8600,
    volume: 10110244,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'INTC',
    companyName: 'Intel Corporation',
    marketCap: 160125239296,
    sector: 'Technology',
    industry: 'Semiconductors',
    beta: 0.6709,
    price: 39.1600,
    lastAnnualDividend: 1.4260,
    volume: 36542703,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'TXN',
    companyName: 'Texas Instruments Incorporated',
    marketCap: 150492266496,
    sector: 'Technology',
    industry: 'Semiconductors',
    beta: 0.9740,
    price: 163.2000,
    lastAnnualDividend: 3.4500,
    volume: 5963901,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  }, {
    symbol: 'AMD',
    companyName: 'Advanced Micro Devices, Inc.',
    marketCap: 141859438592,
    sector: 'Technology',
    industry: 'Semiconductors',
    beta: 1.9408,
    price: 87.5400,
    lastAnnualDividend: 0.0000,
    volume: 106826524,
    exchange: 'NASDAQ Global Select',
    exchangeShortName: 'NASDAQ',
    country: 'US',
    isEtf: false,
    isActivelyTrading: true,
  },
];

export default stocks;
