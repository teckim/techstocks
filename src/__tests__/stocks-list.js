import { screen } from '@testing-library/react';
import StocksList from '../components/StocksList';
import renderWithProviders from '../utils/test-utils';
import { stocks } from '../data/mock';

describe('StocksList components', () => {
  const { asFragment } = renderWithProviders(<StocksList stocks={stocks} />);
  const firstRender = asFragment();

  it('should contain all the items from the mock data', () => {
    const renderedStocks = screen.getByTestId('stocks-list');

    expect(renderedStocks.children.length).toBe(stocks.length);
  });

  it('should render correctly', () => {
    expect(firstRender).toMatchSnapshot();
  });
});
