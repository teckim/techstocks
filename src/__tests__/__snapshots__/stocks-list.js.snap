// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StocksList components should render correctly 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-2"
    data-testid="stocks-list"
  >
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/AAPL"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          AAPL
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Apple Inc.
        </div>
        <div
          class="tracking-wider"
        >
          $ 152.95
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/MSFT"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          MSFT
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Microsoft Corporation
        </div>
        <div
          class="tracking-wider"
        >
          $ 258.83
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/NVDA"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          NVDA
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          NVIDIA Corporation
        </div>
        <div
          class="tracking-wider"
        >
          $ 170.24
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/ASML"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          ASML
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          ASML Holding N.V.
        </div>
        <div
          class="tracking-wider"
        >
          $ 538.97
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/AVGO"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          AVGO
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Broadcom Inc.
        </div>
        <div
          class="tracking-wider"
        >
          $ 513.11
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/CSCO"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          CSCO
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Cisco Systems, Inc.
        </div>
        <div
          class="tracking-wider"
        >
          $ 44.34
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/ADBE"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          ADBE
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Adobe Inc.
        </div>
        <div
          class="tracking-wider"
        >
          $ 391.96
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/QCOM"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          QCOM
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          QUALCOMM Incorporated
        </div>
        <div
          class="tracking-wider"
        >
          $ 153.25
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/INTC"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          INTC
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Intel Corporation
        </div>
        <div
          class="tracking-wider"
        >
          $ 39.16
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/TXN"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          TXN
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Texas Instruments Incorporated
        </div>
        <div
          class="tracking-wider"
        >
          $ 163.2
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="stock-card h-52 flex flex-col px-4 py-6 relative"
      href="/AMD"
    >
      <div
        class="absolute overflow-hidden m-2 inset-0 pt-1"
      >
        <div
          class="[writing-mode:vertical-lr] font-extrabold subpixel-antialiased tracking-widest text-6xl text-white/25 blur-sm"
        >
          AMD
        </div>
      </div>
      <div
        class="text-right mt-auto"
      >
        <div
          class="max-h-20 text-lg uppercase break-normal font-bold tracking-wider text-ellipsis overflow-hidden"
        >
          Advanced Micro Devices, Inc.
        </div>
        <div
          class="tracking-wider"
        >
          $ 87.54
        </div>
      </div>
      <div
        class="p-1 absolute right-4 top-6 rounded-full border border-white"
      >
        <svg
          aria-hidden="true"
          class="h-3 w-3"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 8l4 4m0 0l-4 4m4-4H3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
